cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js

```diff
@@ -0,0 +1,172 @@
+      cy.deleteWorkspaceByName(targetWorkspaceName);
+      sourceWorkspaceId = '';
+      targetWorkspaceId = '';
+    });
+
+    it('Should not list assets from other workspace and generate correct url for inspect url inside a workspace', () => {
+      cy.visit(`${BASE_PATH}/w/${sourceWorkspaceId}/app/objects`);
+      cy.getElementByTestId('savedObjectsTableRowTitle').should('exist');
+      cy.getElementByTestId('savedObjectsTableColumn-workspace_column').should(
+        'not.exist'
+      );
+      cy.contains('opensearch_dashboards_sample_data_ecommerce');
+      // Electron old version may not support search event, so we manually trigger a search event
+      cy.getElementByTestId('savedObjectSearchBar')
+        .type('opensearch_dashboards_sample_data_ecommerce{enter}')
+        .trigger('search');
+      cy.getElementByTestId('savedObjectsTableRowTitle').should(
+        'have.length',
+        1
+      );
+
+      // Click more -> inspect
+      cy.getElementByTestId('euiCollapsedItemActionsButton')
+        .click()
+        .getElementByTestId('savedObjectsTableAction-inspect')
+        .click();
+      cy.location('pathname').should(
+        'include',
+        `/w/${sourceWorkspaceId}/app/indexPatterns`
+      );
+    });
+
+    it('Should list assets from workspaces with permission and generate correct url for inspect url outside workspace', () => {
+      cy.visit(`${BASE_PATH}/app/objects`);
+      cy.getElementByTestId('savedObjectsTableColumn-workspace_column').should(
+        'exist'
+      );
+      // Electron old version may not support search event, so we manually trigger a search event
+      cy.getElementByTestId('savedObjectSearchBar')
+        .type('opensearch_dashboards_sample_data_ecommerce{enter}')
+        .trigger('search');
+      cy.getElementByTestId('savedObjectsTableRowTitle').should(
+        'have.length',
+        2
+      );
+
+      // Find the row belong to the target workspace
+      cy.contains(targetWorkspaceName)
+        .parents('.euiTableRow')
+        // Click more -> inspect
+        .find('[data-test-subj="euiCollapsedItemActionsButton"]')
+        .click()
+        .getElementByTestId('savedObjectsTableAction-inspect')
+        .click();
+      cy.location('pathname').should(
+        'include',
+        `/w/${targetWorkspaceId}/app/indexPatterns`
+      );
+    });
+
+    it('Should not show set as default button when outside workspace', () => {
+      cy.request({
+        url: `${BASE_PATH}/api/opensearch-dashboards/management/saved_objects/_find?workspaces=${targetWorkspaceId}&page=1&type=index-pattern`,
+        headers: {
+          'Osd-Xsrf': 'osd-fetch',
+        },
+      })
+        .then((resp) => {
+          cy.wrap(resp.body.saved_objects).should('have.length', 1);
+          cy.wrap(resp.body.saved_objects[0].id);
+        })
+        .then((indexPatternId) => {
+          cy.visit(`${BASE_PATH}/app/indexPatterns/patterns/${indexPatternId}`);
```

Comment:
we will block index pattern access outside of workspace in future right? can we add a comment here, after this link been blocked, we could remove this this test?

Reply:
Yes, good catch. Comment added.